src/AutoRest.CSharp/LowLevel/Generation/LowLevelExampleComposer.cs

```diff
@@ -0,0 +1,899 @@
+            // in this case, we print the codes of parsing all properties from response
+            ComposeCodeSnippet(clientMethod, methodName, async, false, builder);
+            return builder.ToString();
+        }
+
+        private string GenerateExampleWithRequiredParameters(LowLevelClientMethod clientMethod, string methodName, bool async)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine($"This sample shows how to call {methodName} with required {GenerateParameterAndRequestContentDescription(clientMethod.Signature.Parameters)}{(clientMethod.OperationSchemas.ResponseBodySchema != null ? " and parse the result" : "")}.");
+            // in this case, we print the codes of parsing all properties from response
+            ComposeCodeSnippet(clientMethod, methodName, async, true, builder);
+            return builder.ToString();
+        }
+
+        private bool AreAllParametersOptional(IReadOnlyList<Parameter> parameters)
+        {
+            return !parameters.Any(p => p.DefaultValue == null);
+        }
+
+        /// <summary>
+        /// Check top level properties of the given schema, return true if a required and writable property is found.
+        /// </summary>
+        /// <param name="schema"></param>
+        /// <returns></returns>
+        private bool HasRequiredAndWritablePropertyFromTop(Schema schema)
+        {
+            switch (schema)
+            {
+                case OrSchema o:
+                    foreach (Schema s in o.AnyOf)
+                    {
+                        if (HasRequiredAndWritablePropertyFromTop(s))
+                        {
+                            return true;
+                        }
+                    }
+                    return false;
+                case XorSchema o:
+                    foreach (Schema s in o.OneOf)
+                    {
+                        if (HasRequiredAndWritablePropertyFromTop(s))
+                        {
+                            return true;
+                        }
+                    }
+                    return false;
+                case ObjectSchema o:
+                    // We must also include any properties introduced by our parent chain.
+                    foreach (ObjectSchema s in GetAllSchemaInherited(o))
+                    {
+                        foreach (Property prop in s.Properties)
+                        {
+                            if (prop.IsRequired && !prop.IsReadOnly)
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                    return false;
+            }
+
+            return true;
+        }
+
+        private bool HasOptionalInputValue(IReadOnlyList<Parameter> parameters, Schema? requestSchema)
+        {
+            foreach (var parameter in parameters.SkipLast(1)) // exlucde RequestContext
```

Comment:
We can change to `parameters.Where(p => p.Name != "RequestContext")` to make it more meaningful. I think we should make some assumption (e.g., every protocol method should end with `RequestContext`), otherwise the defence check will be too complex.

Reply:
Same, I think we could remoce the comment.